src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc

```diff
@@ -148,168 +170,184 @@ const std::string LightName(const smart_objects::SmartObject& light_name) {
   return ok ? name : "unknown";
 }
 
-bool CheckLightDataByCapabilities(
+mobile_apis::Result::eType GetItemCapability(
     const smart_objects::SmartObject& capabilities,
-    const smart_objects::SmartObject& light_data) {
+    const std::map<std::string, std::string>& mapping,
+    const std::string& request_parameter,
+    const mobile_apis::Result::eType& switched_off_result) {
+
+  const auto it = mapping.find(request_parameter);
+
+  if (it == mapping.end()) {
+    LOG4CXX_DEBUG(logger_,
+                "Parameter "
+                    << request_parameter
+                    << " doesn't exist in capabilities.");
+    return mobile_apis::Result::UNSUPPORTED_RESOURCE;
+  }
+
+  const std::string& caps_key = it->second;
+  
+  LOG4CXX_DEBUG(logger_,
+                "Checking request parameter "
+                    << request_parameter
+                    << " with capabilities. Appropriate key is " << caps_key);
+
+  if (!capabilities.keyExists(caps_key)) {
+    LOG4CXX_DEBUG(logger_,
+                  "Capability " << caps_key
+                                << " is missed in RemoteControl capabilities");
+    return mobile_apis::Result::UNSUPPORTED_RESOURCE;
+  }
+
+  if (!capabilities[caps_key].asBool()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Capability "
+                      << caps_key
+                      << " is switched off in RemoteControl capabilities");
+    return switched_off_result;
```

Comment:
where switched_off_result is defined? not an item of mobile_apis::Result::xyz

Reply:
@yang1070, switched_off_result has type mobile_apis::Result. 
As you can see above it was passed through function parameters, so value depend on function call context.
That was made for make GetItemCapability more flexible and avoid code duplicating.